src/jdk.internal.jvmstat/share/classes/sun/jvmstat/perfdata/monitor/protocol/local/PerfDataFile.java

```diff
@@ -97,23 +88,9 @@ public class PerfDataFile {
      */
     public static int getLocalVmId(File file) {
         try {
-            // try 1.4.2 and later format first
             return(platSupport.getLocalVmId(file));
         } catch (NumberFormatException e) { }
 
-        // now try the 1.4.1 format
-        String name = file.getName();
-        if (name.startsWith(dirNamePrefix)) {
-            int first = name.indexOf('_');
-            int last = name.lastIndexOf('_');
-            try {
-                if (first == last) {
-                    return Integer.parseInt(name.substring(first + 1));
-                } else {
-                    return Integer.parseInt(name.substring(first + 1, last));
-                }
-            } catch (NumberFormatException e) { }
-        }
         throw new IllegalArgumentException("file name does not match pattern");
```

Comment:
Pre-existing but it would be nice if this actually reported the file and the pattern.

Reply:
Maybe "cannot convert 'filename' to VM id".  (Or to pid.)  We just parse it as an Integer now, so showing the problem filename would be nice.  (Not sure if we would get this far trying a 1.4.1 attach, but if we did, it would help diagnose that.)

Reply:
David and Kevin, thanks for the suggestions. I've update the error message. Does it look OK now?